src/modals/storage-connections/StorageConnectionsDataProvider.js

```diff
@@ -0,0 +1,119 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import getPluginApi from '_/plugin-api'
+import { engineGet, enginePost, enginePut, engineDelete } from '_/utils/fetch'
+import DataProvider from '_/components/helper/DataProvider'
+import { webadminToastTypes } from '_/constants'
+import { msg } from '_/intl-messages'
+
+const createConnection = async (connection) => {
+  return await enginePost(
+    'api/storageconnections',
+    JSON.stringify(connection)
+  )
+}
+
+const editConnection = async (connection, connectionId) => {
+  return await enginePut(
+    `api/storageconnections/${connectionId}`,
+    JSON.stringify(connection)
+  )
+}
+
+const deleteConnection = async (connectionId) => {
+  return await engineDelete(`api/storageconnections/${connectionId}`)
+}
+
+const attachConnection = async (connectionId, domainId) => {
+  return await enginePost(
+    `api/storagedomains/${domainId}/storageconnections`,
+    JSON.stringify({ id: connectionId })
+  )
+}
+
+const detachConnection = async (connectionId, domainId) => {
+  return await engineDelete(`api/storagedomains/${domainId}/storageconnections/${connectionId}`)
+}
+
+const fetchData = async (storageDomainId) => {
+  const allConnectionsJson = await engineGet('api/storageconnections')
+  let storageDomain = await engineGet(`api/storagedomains/${storageDomainId}`)
+  const domainConnectionsJson = await engineGet(`api/storagedomains/${storageDomainId}/storageconnections`)
+
+  const dataCenter = storageDomain.data_centers.data_center[0]
+  if (!dataCenter) {
+    throw new Error('StorageConnectionsDataProvider: failed to fetch data center')
+  }
+
+  storageDomain = await engineGet(`api/datacenters/${dataCenter.id}/storagedomains/${storageDomainId}`)
```

Comment:
we retrieve the storage domain here for the second time - why this is required?

Reply:
@rszwajko It's another endpoint: api/datacenters/${datacenter_id}/storagedomains/${storagedomain_id} vs. api/storagedomains/${storagedomain_id}
Only here the domain status is available

Reply:
IMHO both endpoints ( [/api/storagedomains/{id}](http://ovirt.github.io/ovirt-engine-api-model/master/#services/storage_domains/methods/list) and [/api/datacenters/{id}/storagedomains/{id}](http://ovirt.github.io/ovirt-engine-api-model/master/#services/attached_storage_domains/methods/list)) should return the same - this looks like a bug to me that should be fixed. If not possible please add comment in the code (with bug URL).

Reply:
@rszwajko - Those 2 endpoints do not return the same stuff, and can't.  Some info about a storage domain can only be accessed through the lense of a data center.  ovirt-web-ui has to do the same kind of thing when pulling SD and DC details at login.

Reply:
@sjd78 I haven't dig into this but according to REST API spec they return the same object - http://ovirt.github.io/ovirt-engine-api-model/master/#types/storage_domain
If there are some technical restrictions then that should be reflected in the API - if not this is a bug to me.